File: Java/프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.Arrays;

class Solution {
    public int solution(int[][] targets) {
        int answer = 1;
        Arrays.sort(targets, (a, b) -> Integer.compare(a[1], b[1]));
        int msPoint = targets[0][1];
        for (int[] target : targets) {
            int s = target[0];
            int e = target[1];
            if (s >= msPoint) {
                answer++;
                msPoint = e;
            }
        }
        return answer;
    }
}

File: Swift/프로그래머스/2/181188. 요격 시스템/요격 시스템.swift
import Foundation

func solution(_ targets:[[Int]]) -> Int {
    var targetsC = targets
    targetsC.sort(by: { $0[1] < $1[1] })
    
    var msPoint = targetsC[0][1]
    var answer = 1
    for target in targetsC {
        if target[0] >= msPoint {
            answer += 1
            msPoint = target[1]
        }
    }
    
    return answer
}